auth/realm/base/src/main/java/org/wildfly/security/auth/realm/FileSystemSecurityRealm.java

```diff
@@ -855,7 +886,10 @@ private void writeIdentity(final XMLStreamWriter streamWriter, final LoadedIdent
                             String passwordString;
                             byte[] encoded = BasicPasswordSpecEncoding.encode(password, providers);
 
-                            if (encoded != null) {
+                            if (this.secretKey != null) {
```

Comment:
A very minor comment but just wanted to mention for readability that it's not necessary to use `this` here or below.

Reply:
Removed